### PR TITLE
Fix/connect to

### PR DESCRIPTION
**Pull request by mar-be**
#### Short Description
Fixes https://github.com/OpenTOSCA/container/issues/296

#### Proposed Changes
The interface of a connectTo operation can have the following form: `http://opentosca.org/interfaces/connections/{source|target}/{nodeType}`

- `{source|target}` specifies whether the operation should be only executed on the source or target
- `nodeType` is optional, and it defines the type of the remote node.

Interface names that do not follow this pattern are handled as before.

### CODE REVIEW

**Comment by mar-be** [comment]
Depends on https://github.com/OpenTOSCA/tosca-definitions-test-applications/pull/5

**Comment by mar-be** [comment]
Depends on the latest version of the test https://github.com/OpenTOSCA/tosca-definitions-test-applications/pull/6

File: org.opentosca.planbuilder/org.opentosca.planbuilder.type.plugin.connectsto/src/main/java/org/opentosca/planbuilder/type/plugin/connectsto/bpel/handler/BPELConnectsToPluginHandler.java [hunk 1]
```diff
@@ -126,6 +113,37 @@ private boolean executeConnectsTo(final BPELPlanContext templateContext, final T
...
+                        BPELConnectsToPluginHandler.LOG.debug("Target: " + targetParameterNode.getName());
+                        BPELConnectsToPluginHandler.LOG.debug(iface.getName());
+                        if (iface.getName().matches(".*/source(/[^/]+)?$")) {
+                            BPELConnectsToPluginHandler.LOG.debug("connectTo source defined");
+                            if (!connectToNode.equals(sourceParameterNode)) {
+                                BPELConnectsToPluginHandler.LOG.debug("source does not match");
+                                continue;
+                            }
+                            if (!iface.getName().endsWith("/source")) {
+                                final String targetTypeWithoutVersion = targetParameterNode.getType().getLocalPart().split("_")[0];
```
Comment (nyuuyn) [inline org.opentosca.planbuilder/org.opentosca.planbuilder.type.plugin.connectsto/src/main/java/org/opentosca/planbuilder/type/plugin/connectsto/bpel/handler/BPELConnectsToPluginHandler.java:126]: @lharzenetter is there a better way to check only for the nodetype id/name ? Via WineryVersionUtils and cutting the parts out from the back ?

**Comment by mar-be** [comment]
TODO add documentation

File: docs/components/PlanBuilder.md [hunk 2]
```diff
@@ -91,3 +91,10 @@ Additionally, if the connectTo operation is under an interface with name http://
 1. If B has a start or stop operation under a lifecycle interface, before calling the connectTo operation the stop operation on B is called and after the connectTo operation the start operation is called.
 2. If A has a start or stop operation under a lifecycle interface, before calling the connectTo operation the stop operation on B is called and after the connectTo operation the start operation is called.
 
+To circumvent that the connectTo is invoked multiple times on a target node if it connects to another node as a source node use the following naming convention:
+
+1. Whenever the node is the source node: http://opentosca.org/interfaces/connections/source/{type_without_version}
```
Comment (lharzenetter) [inline docs/components/PlanBuilder.md:96]: Use markdown code and define, whether the type is with a double encoded QName or only it's id/name.

**Comment by lharzenetter** [comment]
Can this be merged now?

**Review by nyuuyn — APPROVED** [review]
LGTM

**Pull request merged**